src/datachain/toolkit/split.py

```diff
@@ -58,10 +58,14 @@ def train_test_split(dc: DataChain, weights: list[float]) -> list[DataChain]:
 
     weights_normalized = [weight / sum(weights) for weight in weights]
 
+    resolution = 2**31 - 1  # Maximum positive value for a 32-bit signed integer.
```

Comment:
I noticed after merging the pull request that my comment is a lie! 🤦🏼 Not that it matters, but the logic is slightly flawed: we should be using `2**31` for the modulo operation and `2**31 - 1` to multiply the normalized weights.